### ISSUE TITLE
Improve `WaywardPredicates` of `Naming/PredicateMethod` cop

### ISSUE THREAD

**Issue by trinistr** [rubocop/rubocop#14823]
## Is your feature request related to a problem? Please describe.

#14327 added `WaywardPredicates` configuration option. However, it's currently not quite there: implementing a Numeric-like interface triggers the cop.

## Describe the solution you'd like

There are two separate things I propose:
1. Add `infinite?` to the default list. `Numeric#infinite?` returns 1, -1, or nil. (`Date#infinite?` returns false, however, but I'm not sure what's the intention there.)
2. Always allow methods with names from `WaywardPredicates`.

For example, this should 1) be considered not a predicate, 2) not trigger the cop:
```ruby
def infinite?
  @value.infinite?
end
```

## Describe alternatives you've considered

1. I could add `infinite?` to my list, but it's a core method, why not in the default one?
2. Again, I could add them to configuration, but methods with these exact names are likely to behave similarly to core methods.

## Additional context

Possibly, `Style/ReturnNilInPredicateMethodDefinition` also should be updated to consider `WaywardPredicates`.
